relay-sampling/src/lib.rs

```diff
@@ -2019,6 +2139,116 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_decaying_rule_precedence() {
+        let rules = SamplingConfig {
+            rules: vec![
+                // Expired
+                SamplingRule {
+                    condition: eq("trace.release", &["1.1.1"], true),
+                    sample_rate: 0.2,
+                    ty: RuleType::Trace,
```

Comment:
This unit test only verifies rules match the context. Do we want to add another test to ensure it works for transactions? I can either duplicate the test + some minor changes, or create a macro for the tests and do some parametrization.

Reply:
Would be nice to test both, how about a parametrized helper function instead of a macro?